django/contrib/postgres/search.py

```diff
@@ -300,3 +300,18 @@ class TrigramSimilarity(TrigramBase):
 class TrigramDistance(TrigramBase):
     function = ''
     arg_joiner = ' <-> '
+
+
+class TrigramWordSimilarity(Func):
+    function = 'WORD_SIMILARITY'
+    output_field = FloatField()
+
+    def __init__(self, expression, string, **extra):
+        if not hasattr(string, 'resolve_expression'):
+            string = Value(string)
+        super().__init__(string, expression, **extra)
```

Comment:
Do we need to swap arguments? IMO we want to keep the same order as in `SIMILARITY()` calls.
```suggestion
class TrigramWordSimilarity(TrigramBase):
    function = 'WORD_SIMILARITY'
```
e.g.
- `TrigramWordSimilarity('Cat sat on mat.', 'cat')` should be equal to `0.30769232` instead of `1`
- `TrigramWordDistance('Cat sat on mat.', 'cat')` should be equal to `0.6923077` instead of `0`

Reply:
Paolo, Can you take a look? (\cc @pauloxnet) :point_up:

Reply:
This is a good question. This thing is that the arguments of `SIMILARITY()` are commutative where for `WORD_SIMILARITY()` they are not.

The rationale for the argument swap is that i wanted to keep the column/field name on the left of the Django call. But most often when making a full-text search, the "long" string (`'Cat sat on mat.'`) is in the column where the "short" one (`'cat'`) is an argument so it will mostly be called like `WORD_SIMILARITY(a_column, 'the argument')` and in the Django call, i wanted to keep the column name as the first argument like in `TrigramSimilarity`.

I don't know what is best, writing the patch with the arguments in the other order is easy too, i'll follow your decision :)

Reply:
>  .. i'll follow your decision :)

Just asking, I'm not an expert :shrug:. We can wait for the second opinion from Paolo.

Reply:
@hannseman @jamesturk Can I ask for your opinion? :point_up:

Reply:
I think it's probably for the best to try and keep the order of the arguments the same as in PostgreSQL. Feels like people will be cross-referencing against the PostgreSQL documentation when using this and swapping the arguments could be confusing.

Reply:
As far I'm aware if we want to keep the same order as in PostgreSQL, it should be :thinking: 
```python
TrigramWordSimilarity(string, expression)
```
For example
```python
TrigramWordSimilarity('cat', 'Cat sat on mat.')
```

Reply:
@felixxm yes that's what I'm thinking.

Reply:
> Paolo, Can you take a look? (\cc @pauloxnet) point_up

Sorry, I totally missed the notification of this. I'll take a look

Reply:
I'd like to keep this moving so i still have a chance to get this in time for the feature freeze for 4.0 🤞 

I'll change this PR to have 
```python
TrigramWordSimilarity(string, expression)
```
And keep the same order in SQL.